### DISCUSSION TITLE
Models with limb-darkening coefficient

### DISCUSSION THREAD

**Discussion by IvaPIvanova** [rpoleski/MulensModel#97] [Q&A]
I'm using MulensModel with a set of simulated data to generate light curves for single-lens microlensing events. I've created both PSPL models and finite source models, and am now looking to update them by introducing limb-darkening. I have a linear limb-darkening coefficient for each star, which is an output from the simulation itself (the code handles computation of linear limb-darkening coefficients using precomputed spline interpolates fitted to data). We cover a range of values for rho so I've successfully used the code below thus far:

if rho < 0.001:
    model_finitesource.set_magnification_methods([t_1,'point_source',t_2])
elif rho <0.1:
    model_finitesource.set_magnification_methods([t_1,'finite_source_uniform_Gould94',t_2])
else:
    model_finitesource.set_magnification_methods([t_1,'finite_source_uniform_Lee09',t_2])
    
However, I'm struggling to switch to using finite_source_LD_Yoo04  (instead of finite_source_uniform_Gould94) and finite_source_LD_Lee09 (instead of finite_source_uniform_Lee09) with a pre-defined limb-darkening coefficient as a variable in the script. I'm not sure on the correct syntax of .set_magnification_methods() in my case. I tried following [example_25_LD.py ](https://github.com/rpoleski/MulensModel/blob/40688076314607000e827ce5def30c7bac1018f7/examples/example_25_LD.py) to adapt with my code but was unsuccessful (I get the TypeError: __init__() got an unexpected keyword argument 'magnification_methods' at line uniform_model = mm.Model( )

I would highly appreciate some help with implementing pre-computed LDC  with the finite_source_LD_Yoo04  and finite_source_uniform_Lee09 models.

**Comment by rpoleski** [comment]
Hi,

> However, I'm struggling to switch to using finite_source_LD_Yoo04 (instead of finite_source_uniform_Gould94)

It's not clear what the problem is. Your `Model.set_magnification_methods()` code seems fine. You just need to change the method name and provide limb darkening coefficient using [`Model.set_limb_coeff_gamma()`](https://rpoleski.github.io/MulensModel/MulensModel.model.html#MulensModel.model.Model.set_limb_coeff_gamma) or [`Model.set_limb_coeff_u()`](https://rpoleski.github.io/MulensModel/MulensModel.model.html#MulensModel.model.Model.set_limb_coeff_u) depending on the convention you use. If these explanations are not enough, then please copy-paste example code that fails.
You later link example 25, which was considered at some point, but is not among examples now (see [`examples/` dir](https://github.com/rpoleski/MulensModel/tree/master/examples)).

**Comment by IvaPIvanova** [comment]
Hi, thank you so much for your quick response. I had just gotten myself with that example 25 confused on how to set the LDC. All good now. Thanks for your help!
